File: clean_queues.py
#!/usr/bin/env python3
import pika
import sys

def clean_queues():
    print("🧹 Connecting to RabbitMQ...\n")
    
    credentials = pika.PlainCredentials('guest', 'guest')
    connection = pika.BlockingConnection(
        pika.ConnectionParameters('localhost', 5672, '/', credentials)
    )
    channel = connection.channel()
    
    queues = ['travel.bookings', 'travel.admin']
    
    print("Purging queues...\n")
    for queue in queues:
        try:
            # Declare passive to check if queue exists and get message count
            method_frame = channel.queue_declare(queue=queue, passive=True)
            message_count = method_frame.method.message_count
            
            if message_count > 0:
                channel.queue_purge(queue)
                print(f"  ✅ Purged {message_count} messages from: {queue}")
            else:
                print(f"  ℹ️  Queue '{queue}' is empty")
        except Exception as e:
            print(f"  ❌ Error with '{queue}': {str(e)}")
    
    connection.close()
    print("\n✅ Queue cleanup complete!")

if __name__ == '__main__':
    clean_queues()

File: Travelling/CleanQueues.cs
using RabbitMQ.Client;

class Program
{
    static void Main(string[] args)
    {
        var factory = new ConnectionFactory
        {
            HostName = "localhost",
            Port = 5672,
            UserName = "guest",
            Password = "guest",
            VirtualHost = "/"
        };

        using var connection = factory.CreateConnection();
        using var channel = connection.CreateModel();

        Console.WriteLine("🧹 Cleaning RabbitMQ Queues...\n");

        PurgeQueue(channel, "travel.bookings");
        PurgeQueue(channel, "travel.admin");

        Console.WriteLine("\n✅ Queue cleanup complete!");
    }

    static void PurgeQueue(IModel channel, string queueName)
    {
        try
        {
            var declareOk = channel.QueueDeclarePassive(queueName);
            if (declareOk.MessageCount > 0)
            {
                channel.QueuePurge(queueName);
                Console.WriteLine($"  ✅ Purged {declareOk.MessageCount} messages from: {queueName}");
            }
            else
            {
                Console.WriteLine($"  ℹ️  Queue '{queueName}' is empty");
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine($"  ❌ Error with '{queueName}': {ex.Message}");
        }
    }
}